powerauth-java-crypto/src/main/java/io/getlime/security/powerauth/crypto/lib/generator/KeyGenerator.java

```diff
@@ -187,6 +187,32 @@ public SecretKey deriveSecretKeyHmac(SecretKey secret, byte[] index) throws Gene
         return keyConvertor.convertBytesToSharedSecretKey(newKeyBytes);
     }
 
+    /**
+     * Derive a new secret key KEY_SHARED from a master secret key KEY_MASTER
+     * based on following KDF:
+     * <pre>
+     *    BYTES = index, total 16 bytes
+     *    KEY_SHARED[BYTES] = 32B_TO_16B(HMAC-SHA256(BYTES, KEY_MASTER))
+     * </pre>
+     *
+     * This function is similar to {@link #deriveSecretKeyHmac(SecretKey, byte[])}, but calculates the derivation in
+     * the correct way and as it's specified in our documentation (look for pseudo function {@code KDF_INTERNAL.derive()}).
+     *
+     * @param secret A master shared key.
+     * @param index A byte array index of the key.
+     * @return A new derived key from a master key with given index.
+     * @throws GenericCryptoException In case key derivation fails.
+     * @throws CryptoProviderException In case cryptography provider is incorrectly initialized.
+     */
+    public SecretKey deriveSecretKeyHmacInternal(SecretKey secret, byte[] index) throws GenericCryptoException, CryptoProviderException {
```

Comment:
This is a temporary solution. This PR suppose to implement IV for the status blob encryption, so let's keep it simple. If I fix the original method, then as a part of the task, I'll need to solve all backward compatibility, just because we made a mistake in the past. My intention here, was to implement IV first, then fix that incorrect derivation function in a separate PRs.

Reply:
Ok.